apps/stats-dapp/src/containers/KeyStatusCardContainer/KeyStatusCardContainer.tsx

```diff
@@ -1,57 +1,47 @@
 import { KeyStatusCard } from '@webb-tools/webb-ui-components/components/KeyStatusCard';
-import { KeyStatusCardProps } from '@webb-tools/webb-ui-components/components/KeyStatusCard/types';
-import { Spinner } from '@webb-tools/icons';
-import React, { FC, forwardRef, useMemo } from 'react';
-import { KeyStatusCardContainerProps } from './types';
 import { useStatsContext } from '../../provider/stats-provider';
-import { PublicKey, useActiveKeys } from '../../provider/hooks';
+import { useBlocks } from '../../provider/hooks';
+import { useMemo } from 'react';
+
 /**
  * The wrapper of UI component. Handle logic and mapping fields between backend API and component API
  */
 export const KeyStatusCardContainer = () => {
-  const { dkgDataFromPolkadotAPI, sessionHeight } = useStatsContext();
-
-  const { error, isFailed, isLoading, val: data } = useActiveKeys();
-
-  const { currentKey } = useMemo<{
-    currentKey: PublicKey | null | undefined;
-  }>(() => {
-    return {
-      currentKey: data ? data[0] : null,
-    };
-  }, [data]);
+  const {
+    dkgDataFromPolkadotAPI: {
+      currentSessionNumber,
+      currentKey,
+      currentSessionTimeFrame: { start: sessionStart, end: sessionEnd },
+      currentAuthorities,
+    },
+  } = useStatsContext();
 
   const { time } = useStatsContext();
 
-  const authorities = useMemo(
-    () =>
-      (currentKey?.keyGenAuthorities ?? []).reduce((acc, cur) => {
-        acc.add(cur);
-        return acc;
-      }, new Set() as KeyStatusCardProps['authorities']) ?? new Set<string>(),
-    [data]
-  );
+  const { val: blocksData } = useBlocks();
 
-  const activeKeyData = useMemo(() => {
-    return {
-      key: currentKey?.compressed,
-      session: Number(currentKey?.session),
-    };
-  }, [data, currentKey]);
+  const showDetails = useMemo(() => {
+    if (blocksData?.finalized - blocksData?.latestIndexedBlock < 5) {
```

Comment:
Before performing calculations on two potentially undefined values, you should first check if either value is undefined.

```tsx
if (blocksData && blocksData.finalized - blocksData.lastestIndexedBlock < 5)
```